api/v1beta1/nutanixmachine_types.go

```diff
@@ -61,13 +61,12 @@ type NutanixMachineSpec struct {
 	// of the Prism Central), in which the Machine's VM will be created.
 	// The cluster identifier (uuid or name) can be obtained from the Prism Central console
 	// or using the prism_central API.
-	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Optional
```

Comment:
same as above, why optional

Reply:
We need it for backward compatibility. This attribute should only be defined if failure domains are not configured on the NutanixCluster object.